### ISSUE TITLE
Compose box for topic (and DM) narrows

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#144]
Splitting this out from:
* #72 

When we originally added a compose box a few months ago (#9), we consciously deferred trying to structure the code in a way that would work well for the several different kinds of narrows we'll want a compose box for, and just wrote one specifically for stream narrows, the most complex case. That's because at the time our focus was on derisking Flutter, and so what we wanted from a compose box was a context in which to experiment with text inputs, their interaction through controllers and focus nodes, and picking files and taking photos to attach.

For an app for people to actually use, though, even in alpha, we'll want a compose box in those other narrows. This will call for some refactoring of the compose-box code, which is why I'm splitting it out from #72.

Once that's done and we have a working compose box for topic narrows, it should be easy to add one for DM narrows too, provided those exist in the first place (#142). So if this happens after #142, it'll include DM narrows as well as topic narrows; if it happens before, we can add the compose box there as part of #142.
